noir-projects/aztec-nr/aztec/src/encrypted_logs/log_assembly_strategies/default_aes128/event.nr

```diff
@@ -233,7 +266,10 @@ where
     offset += log_bytes_as_fields.len();
 
     for i in offset..PRIVATE_LOG_SIZE_IN_FIELDS {
-        final_log[i] = unsafe { random() };
+        // We need to get a random value that fits in 31 bytes to not leak information about the size of the log
+        // (all the "real" log fields contain at most 31 bytes because of the way we convert the bytes to fields).
+        // Safety: randomness cannot be constrained.
+        final_log[i] = unsafe { random_with_max_byte_size::<31>() };
```

Comment:
@iAmMichaelConnor we leaked here the information of how long the log actually is so I fixed it.

Reply:
Oh, this isn't the correct fix to apply here. We want this log to be indistinguishable from all logs of _all log assembly strategies_. So all of the fields of a log should be _full_ fields; not truncated to 31 bytes. That's why I wrote (but didn't plumb-in) the `be_bytes_to_padded_fields` fn in `src/utils/bytes.nr`: It's a bit constraint-heavy, but it pads all 31-byte fields with 6 more bits of randomness, so that all of the log fields are 254-bit random-looking values, instead of 248-bit random-looking values.

Reply:
E.g. if you consider someone encrypting with poseidon2 instead of AES, well poseidon2 spits out full field elements.

Reply:
Oh ok, this would be a lovely comment to put in your original code 😜

>but it pads all 31-byte fields with 6 more bits of randomness

Given that a field is 253.6 bits and not 254 I think those missing 0.4 bits would still nuke your privacy set.

Reply:
Created an [issue](https://github.com/AztecProtocol/aztec-packages/issues/12749) for this and linked it in the code.